File: 15-3sum/15-3sum.java
class Solution {
    public List<List<Integer>> threeSum(int[] nums) {
        Arrays.sort(nums);
        List<List<Integer>> answer = new ArrayList<List<Integer>>();
        Set<List<Integer>> hashy = new HashSet<>();
        for(int i = 0;i< nums.length;i++)
        {
            int left = i+1;
            int right = nums.length -1;
            int target = -nums[i];
             System.out.println(target);
            while(left < right)
            {
               int current = nums[left] + nums[right];
               
                if( current == target)
                {
                    List<Integer> temp = new ArrayList<Integer>();
                    temp.add(nums[i]);
                    temp.add(nums[left]);
                    temp.add(nums[right]);
                    if(!hashy.contains(temp))
                    {
                        answer.add(temp);
                        hashy.add(temp);
                    }
                   
                    right--;
                }
                else if(current > target)
                {
                    right--;
                }
                else
                {
                    left++;
                }
            }
        }
        return answer;
    }
}

File: Antra_C#_2/15-3sum/15-3sum.cs
public class Solution {
    public IList<IList<int>> ThreeSum(int[] newArr) {
        //two pointer approach
     //   HashSet<IList<int>> hash1 = new HashSet<IList<int>>();
        List<IList<int>> answer = new List<IList<int>>();
        
        for(int i =0;i< newArr.Length;i++)
        {
            Console.WriteLine(newArr[i]);
        }
        Array.Sort(newArr);
        for(int i =0;i< newArr.Length;i++)
        {
            if(i != 0)
            {
                if(newArr[i] == newArr[i-1])
                {
                    continue;
                }
            }
           int j = i+1;
           int k = newArr.Length - 1;
            while(j<k)
            {
                if(newArr[i] + newArr[j] + newArr[k]== 0)
                {
                    answer.Add(new List<int>{newArr[i],newArr[j],newArr[k]});
                    j++;
                    while(newArr[j] == newArr[j-1] && j < k)
                    {
                        j++;
                    }
                }
                else if(newArr[i] + newArr[j] + newArr[k]< 0)
                {
                    j++;
                }
                else
                {
                    k--;
                }
            }
        }
        //Console.WriteLine(hash1);
        return answer;
    }
}